resolwe_bio/processes/import_data/seq_reads.yml

```diff
@@ -218,3 +218,248 @@
+      do
+        READS_NAME=`basename "${report_dir}" _fastqc`
+        ENCODING=`parse_encoding_type.py "${report_dir}fastqc_data.txt"`
+
+        if [ "${ENCODING}" == "Illumina 1.5" ] ||  [ "${ENCODING}" == "Illumina 1.3" ]; then
+
+          echo "Recoding input reads from Phred64 encoding to Phred33 encoding."
+          mv "${READS_NAME}.fastq" input_reads.fastq
+          TrimmomaticSE input_reads.fastq reformated.fastq.gz TOPHRED33
+          mv reformated.fastq.gz "${READS_NAME}.fastq.gz"
+
+        elif [ "${ENCODING}" != "Sanger / Illumina 1.9" ]; then
+          re-error "Only Sanger / Illumina 1.9 / llumina 1.5 / Illumina 1.3 encoding is supported."
+        fi
+      done
+
+      FASTQGZ=(*.fastq.gz)
+      re-save-file-list fastq "${FASTQGZ[@]}"
+
+      FASTQC=(*_fastqc.zip)
+      FASTQC_URL=()
+
+      for fqc in "${FASTQC[@]}"
+      do
+        NAME=`basename "$fqc" _fastqc.zip`
+        FASTQC_URL+=("{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}")
+      done
+
+      re-save-file-list fastqc_archive "${FASTQC[@]}"
+      re-save-list fastqc_url "${FASTQC_URL[@]}"
+
+- slug: basespace-fastq-paired
+  name: BaseSpace Reads (paired-end)
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: resolwebio/legacy:latest
+    resources:
+      network: true
+      secrets: true
+  data_name: 'BaseSpace {{ upstream_file_ids.0|default("?") }}'
+  version: 0.0.1
+  type: data:reads:fastq:paired
+  category: upload
+  flow_collection: sample
+  persistence: RAW
+  description: >
+    Import paired-end reads in FASTQ format from BaseSpace.
+  input:
+    - name: upstream_file_ids
+      label: BaseSpace NGS upstream read (FASTQ) file IDs
+      type: list:basic:string
+    - name: downstream_file_ids
+      label: BaseSpace NGS downstream read (FASTQ) file IDs
+      type: list:basic:string
+    - name: access_token_secret
+      label: BaseSpace access token
+      type: basic:secret
+      description: >
+        BaseSpace access token secret handle needed to download the files.
+  output:
+    - name: fastq
+      label: Reads file (mate 1)
+      type: list:basic:file
+    - name: fastq2
+      label: Reads file (mate 2)
+      type: list:basic:file
+    - name: fastqc_url
+      label: Quality control with FastQC (Upstream)
+      type: list:basic:file:html
+    - name: fastqc_url2
+      label: Quality control with FastQC (Downstream)
+      type: list:basic:file:html
+    - name: fastqc_archive
+      label: Download FastQC archive (Upstream)
+      type: list:basic:file
+    - name: fastqc_archive2
+      label: Download FastQC archive (Downstream)
+      type: list:basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+
+      function extract_filename_from_output() {
+        echo $1 | sed -nE 's/^filename=(.+)/\1/p'
+      }
+
+      SECRET_PATH=/secrets/{{ access_token_secret.handle }}
+
+      FW_FASTQ=()
```

Comment:
Reused from https://github.com/genialis/resolwe-bio/blob/23e9deb50a8a972b544187a92b0180488b46a7b8/resolwe_bio/processes/import_data/seq_reads.yml#L145

Maybe it would be good idea to encapsulate this functionality too. cc @jkokosar